model/Tree.java

```diff
@@ -1,5 +1,38 @@
 package model;
 
-public class Tree {
-    
+import impresario.IModel;
+import impresario.IView;
+
+import javafx.scene.Scene;
+import userinterface.View;
+import userinterface.ViewFactory;
+
+public class Tree extends TLC implements IView, IModel  {
```

Comment:
should extend EntityBase rather than TLC only needs to implement IModel:
public class Tree extends EntityBase implements Imodel {...}